src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-file-history/jj-file-history.component.ts

```diff
@@ -32,34 +34,29 @@ export class JJFileHistoryComponent implements OnInit {
     private historyRecordService: HistoryRecordService
   ) {
     this.historyRecordService.refreshFileHistory.subscribe(ticketNumber => {
-      this.ticketNumber = ticketNumber;
       this.dataSource = new MatTableDataSource<HistoryRecord>();
-      this.getAllFileHistory();
+      this.getAllHistories();
     });
   }
 
   ngOnInit(): void {
-    this.getAllFileHistory();
+    this.getAllHistories();
   }
 
-  getAllFileHistory() {
-    this.historyRecordService.getFileHistories(this.ticketNumber).subscribe((response: FileHistory[]) => {
-      this.logger.info('FileHistoryComponent::getAllFileHistory', response)
-      this.fileHistory = response;
-      this.getAllEmailHistory();
-    });
-  }
-
-  getAllEmailHistory() {
-    this.historyRecordService.getEmailHistories(this.ticketNumber).subscribe((response: EmailHistory[]) => {
-      this.logger.info('FileHistoryComponent::getAllEmailHistory', response)
-      this.emailHistory = response;
+  getAllHistories() {
+    let observables = [];
+    observables.push(this.historyRecordService.getFileHistories(this.ticketNumber))
+    observables.push(this.historyRecordService.getEmailHistories(this.ticketNumber))
+    
+    forkJoin(observables).subscribe((responses: any[]) => {
```

Comment:
I've been learning that whenever there is a .subscribe on an Observable, there must be an .unsubscribe on destroy or there will be memory leaks. Is this right?

Reply:
I think it would be good, yes. This was just one article, but I've read that in various places elsewhere.